src/em/prov/easyconnect/ec_enrollee.cpp

```diff
@@ -1839,7 +1839,7 @@ bool ec_enrollee_t::check_bss_info_has_cce(const wifi_bss_info_t& bss_info) {
     };
 
 
-    if (!bss_info.ie || bss_info.ie_len == 0) {
+    if (bss_info.ie_len == 0) {
```

Comment:
Can you keep the NULL check?
```suggestion
    if (bss_info.ie == NULL || bss_info.ie_len == 0) {
```

Reply:
I tried this change but still there is a warning as below:

/home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../..//src/em/prov/easyconnect/ec_enrollee.cpp: In member function ‘bool ec_enrollee_t::check_bss_info_has_cce(const wifi_bss_info_t&)’:
/home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../..//src/em/prov/easyconnect/ec_enrollee.cpp:1842:21: warning: the address of ‘wifi_bss_info_t::ie’ will never be NULL [-Waddress]
 1842 |     if (bss_info.ie == NULL || bss_info.ie_len == 0) {
      |                     ^
In file included from /home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../../../halinterface/include/wifi_hal.h:83,
                 from /home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../../../OneWifi//include/wifi_base.h:30,
                 from /home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../../../OneWifi//include/wifi_webconfig.h:29,
                 from /home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../..//inc/em_base.h:32,
                 from /home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../..//inc/ec_enrollee.h:4,
                 from /home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../..//src/em/prov/easyconnect/ec_enrollee.cpp:1:
/home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../../../halinterface/include/wifi_hal_generic.h:364:19: note: ‘wifi_bss_info_t::ie’ declared here
  364 |     unsigned char ie[512]; /**< Information elements. */
      |                   ^~
g++ -o /home/amarnath/easymesh_project/unified-wifi-mesh/build/ctrl/../..//install/bin/onewifi_em_ctrl /home/amarnath/easymesh_project/unified-wifi-mesh/build/ct

Since the halinterface defines ie as static structure, there is no case where in the ie would be NULL. So, removing this check probably makes sense

Reply:
Makes sense, looks good to me